cmd/rpc.go

```diff
@@ -181,18 +190,19 @@ func (client *RPCClient) Call(serviceMethod string, args interface {
 }, reply interface{}) (err error) {
 	lockedCall := func() error {
 		client.RLock()
-		defer client.RUnlock()
-
-		if client.retryTicker != nil {
+		retryTicker := client.retryTicker
+		authToken := client.authToken
+		client.RUnlock()
+		if retryTicker != nil {
 			select {
-			case <-client.retryTicker.C:
+			case <-retryTicker.C:
 			default:
 				return errRPCRetry
 			}
 		}
 
 		// Make RPC call.
-		args.SetAuthArgs(AuthArgs{client.authToken, client.args.RPCVersion, time.Now().UTC()})
+		args.SetAuthArgs(AuthArgs{authToken, client.args.RPCVersion, time.Now().UTC()})
```

Comment:
There is a chance of not using correct token here which is just updated when waiting on `retryTicker.C`.

You would need to retrieve when its needed.